lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -0,0 +1,205 @@
+        /// </summary>
+        public Stream Open(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlockBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Create(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlockBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// Deletes a CloudBlockBlob
+        /// <see cref="IFileSystem.Delete"/>
+        /// </summary>
+        public void Delete(Uri fileUri)
+        {
+            _client.GetBlockBlobReference(fileUri).Delete();
+        }
+
+        /// <summary>
+        /// Checks if a CloudBlockBlob exists.
+        /// <see cref="IFileSystem.Exists"/>
+        /// </summary>
+        public bool Exists(Uri fileUri)
+        {
+            return _client.GetBlockBlobReference(fileUri).Exists();
+        }
+
+        /// <summary>
+        /// Copies to a CloudBlockBlob from another CloudBlockBlob.
+        /// <see cref="IFileSystem.Copy"/>
+        /// </summary>
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            _client.GetBlockBlobReference(destinationUri).StartCopyFromBlobAsync(sourceUri).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Copies from a CloudBlockBlob to a local file.
+        /// <see cref="IFileSystem.CopyToLocal"/>
+        /// </summary>
+        public void CopyToLocal(Uri remoteFileUri, string localName)
+        {
+            _client.GetBlockBlobReference(remoteFileUri).DownloadToFile(localName, FileMode.CreateNew);
+        }
+
+        /// <summary>
+        // Copies to a CloudBlockBlob from a local file.
+        /// <see cref="IFileSystem.CopyFromLocal"/>
+        /// </summary>
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            _client.GetBlockBlobReference(remoteFileUri).UploadFromFile(localFileName, FileMode.Open);
+        }
+
+        /// <summary>
+        /// Azure blobs does not have the sense of the existence of a "directory."
+        /// We will not throw an error, but we will not do anything either.
+        /// </summary>
+        public void CreateDirectory(Uri directoryUri)
+        {
+        }
+
+        /// <summary>
+        /// Recursively deletes blobs under a specified "directory URI."
+        /// If only the container is specified, the entire container is deleted.
+        /// </summary>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            var uriSplit = directoryUri.AbsolutePath.Split(new[] {"/"}, StringSplitOptions.RemoveEmptyEntries);
+            if (!uriSplit.Any())
+            {
+                throw new ArgumentOutOfRangeException(@"URI must contain at least the container");
```

Comment:
This exception type is artifact of implementation detail. Use exception type that reflects the semantics.